packages/fuel-gauge/src/script-main-args.test.ts

```diff
@@ -0,0 +1,82 @@
+import { generateTestWallet } from '@fuel-ts/wallet/test-utils';
+import { readFileSync } from 'fs';
+import type { BigNumberish } from 'fuels';
+import { NativeAssetId, Provider, bn, ScriptFactory } from 'fuels';
+import { join } from 'path';
+
+import scriptAbi from '../test-projects/script-main-args/out/debug/script-main-args-abi.json';
+
+import { getScript } from './utils';
+
+const scriptBin = readFileSync(
+  join(__dirname, '../test-projects/script-main-args/out/debug/script-main-args.bin')
+);
+
+const setup = async () => {
+  const provider = new Provider('http://127.0.0.1:4000/graphql');
+
+  // Create wallet
+  const wallet = await generateTestWallet(provider, [[5_000_000, NativeAssetId]]);
+
+  return wallet;
+};
+
+type Baz = {
+  x: number;
+};
+
+describe('Script Coverage', () => {
+  it('can call script and use main arguments', async () => {
+    const wallet = await setup();
+    // #region typedoc:script-call-factory
+    const scriptInstance = new ScriptFactory<BigNumberish>(scriptBin, scriptAbi).script;
+    const foo = 33;
+
+    const { value, logs } = await wallet.callScript<BigNumberish[], BigNumberish>(scriptInstance, [
```

Comment:
I think this types should follow the types of [InvokeFunction](https://github.com/FuelLabs/fuels-ts/blob/cef77013e14461b07a525213046859ba1013dd29/packages/contract/src/types.ts#L34) it will help too keep consistency and I think it what you are looking for.

```suggestion
    const { value, logs } = await wallet.callScript<BigNumberish[], BigNumberish>(scriptInstance, [
```